test/e2e/webhelp.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { expect, test } from '@playwright/test';
+import { testSelector } from '../utils';
+import { WebHelpPage } from '../page/webhelp-page';
+import {  RESOLUTIONS } from '../visual/visual-constants';
+
+test.describe('WebHelp page interactivity', async () => {
+    test.beforeEach(async ({ page }) => {
+        const webHelpPage = new WebHelpPage(page, '/docs/test-page.html');
+        await webHelpPage.init();
+    });
+
+    test(`Should open page  item  properly on desktop`, async ({ page }) => {
+        await page.setViewportSize(RESOLUTIONS[0]);
+        const element = page.locator('nav').locator('ul.toc').first();
+        const tocItem = element.locator(testSelector('toc-item')).filter({ hasText: 'Basics' }).first();
+        await tocItem.click();
+        const linkTocItem = element.locator('a[href="basic-syntax.html"]').first();
+        await linkTocItem.click();
+        expect(page.url()).toContain('basic-syntax.html');
+    });
+
+
+    test(`Should copy codeblock content when copy button clicked`, async ({ page }) => {
+        const EXPECTED_CODEBLOCK_CONTENT = `plugins {
+kotlin("kapt") version "1.9.22"
+}`;
+        const codeBlock = page.locator(testSelector('code-block')).first();
+        await codeBlock.hover();
+        const copyButton = codeBlock.locator(testSelector('copy-button')).first();
+        await copyButton.click();
+        // since Playwright doesn't have access to the clipboard, we have to test it manually by creating a textarea and pasting the clipboard content into it
```

Comment:
Have you tried `grantPermissions`?
```javascript
await context.grantPermissions(["clipboard-read"]);
const handle = await page.evaluateHandle(() => navigator.clipboard.readText());
const clipboardContent = await handle.jsonValue();
```

Reply:
I've tried this and made this work locally but not in Docker unfortunately